File: teste1.py
def escrever_no_arquivo(nome_arquivo, *parametros):
    with open(nome_arquivo, "w") as arquivo:
        for parametro in parametros:
            arquivo.write(str(parametro) + "\n")
    print(f"Parâmetros escritos em '{nome_arquivo}' com sucesso!")


nome_arquivo = input("Digite o nome do arquivo: ")
parametros = input("Digite os parâmetros separados por espaço: ").split()

escrever_no_arquivo(nome_arquivo,*parametros)

File: teste1.c
#include <stdio.h>

void escrever_no_arquivo(const char *nome_arquivo, int argc, char *argv[]) {
    FILE *arquivo = fopen(nome_arquivo, "w");
    if (arquivo == NULL) {
        printf("Erro ao abrir o arquivo.\n");
        return;
    }

    for (int i = 2; i < argc; i++) {
        fprintf(arquivo, "%s\n", argv[i]);
    }

    fclose(arquivo);
    printf("Parâmetros escritos em '%s' com sucesso!\n", nome_arquivo);
}

int main(int argc, char *argv[]) {
    if (argc < 3) {
        printf("Uso: %s <nome_arquivo> <parametro1> <parametro2> ...\n", argv[0]);
        return 1;
    }

    escrever_no_arquivo(argv[1], argc, argv);
    return 0;
}